lib/oauth2-client.js

```diff
@@ -106,7 +100,7 @@ export async function getNewAccessToken({
     const access_token = await _requestAccessToken(
       {url: token_endpoint, body, maxRetries});
     if(access_token) {
-      return {client_id, token_endpoint, access_token};
+      return access_token;
```

Comment:
This looks like a breaking change to a public API which I don't see listed as a breaking change in the change log.  What is the inspiration behind this change?

Was this a bug because the return value here was in variance with the return value on L163?

Per our discussions earlier this week, it is often desirable to return an extensible object from public APIs instead of primatives like strings/booleans etc.  We might return `{accessToken}` today and `{accessToken, foo}` tomorrow.  If we just return `accessToken` and need to convert from a string to an object, it's a breaking change.

Reply:
Previously this internal function returned an object so that information could be saved to the database, but the overall function itself returned a string: https://github.com/digitalbazaar/bedrock-oauth2-client/blob/main/lib/oauth2-client.js#L79 and that is noted in the documentation: https://github.com/digitalbazaar/bedrock-oauth2-client/blob/main/lib/oauth2-client.js#L41

Since we are already making a breaking change, should it be updated to an object?

Reply:
You've referred to something as an `internal function`.  This `getNewAccessToken` function is part of the public API right?

I see that my comment about L163 is incorrect because that is *not* part of this function.  Sorry to complicate the conversation.  :(

So, the return value her was `{client_id, token_endpoint, access_token}`, and changing it to `{access_token}` could be done as a non-breaking change right?  So the return value was *already* an object, correct?

Reply:
@mattcollier so there was a bug here, 2 different places were calling `getNewAccessToken` and expecting different values. When handling the token error, it was expecting the string: https://github.com/digitalbazaar/bedrock-oauth2-client/blob/main/lib/authzHttpClient.js#L89, which follows the same pattern as `getAccessToken` https://github.com/digitalbazaar/bedrock-oauth2-client/blob/main/lib/authzHttpClient.js#L44, but during the now-removed function `_getUncachedAccessToken` the object was converted into a string. https://github.com/digitalbazaar/bedrock-oauth2-client/blob/main/lib/oauth2-client.js#L67

Reply:
@mattcollier Updated.